scripts/machine_translation/train_transformer.py

```diff
@@ -325,24 +341,33 @@ def train(args):
                             {'learning_rate': args.lr, 'beta1': 0.9,
                              'beta2': 0.98, 'epsilon': 1e-9, 'lr_scheduler': lr_scheduler})
     # Load Data
-    if args.bucket_scheme == 'constant':
-        bucket_scheme = ConstWidthBucket()
-    elif args.bucket_scheme == 'linear':
-        bucket_scheme = LinearWidthBucket()
-    elif args.bucket_scheme == 'exp':
-        bucket_scheme = ExpWidthBucket(bucket_len_step=1.2)
+    if args.sampler == 'BoundedBudgetSampler':
+        train_batch_sampler = BoundedBudgetSampler(lengths=[(ele[2], ele[3]) for ele in data_train],
```

Comment:
In Fairseq, they use the max of source and target: https://fairseq.readthedocs.io/en/latest/_modules/fairseq/data/language_pair_dataset.html#LanguagePairDataset.num_tokens

Reply:
OK, I find that it's handled in the BoundedBudgetSampler.